friends/src/App.js

```diff
@@ -0,0 +1,70 @@
+
+import React, { Component } from 'react';
+import './App.css';
+import { connect } from 'react-redux';
+import { getFriends, addFriend, deleteFriend, updateFriend } from './actions'
+import Friends from './components/Friends';
+import FriendForm from './components/CreateFriendForm';
+import FriendUpdateForm from './components/UpdateFriendForm';
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isUpdate: false,
+      upDatingFriend: {}
+    }
+  }
+
+  componentDidMount() {
+    this.props.getFriends()
+  }
+
+  handleUpdate = event => {
+    this.setState({
+      isUpdate: true,
+      upDatingFriend: this.props.friends.filter(friend => friend.id === event.target.id)[0]
+    })
+  }
+
+  updateSubmit = friend => {
+    this.setState({
+      isUpdate: false,
+      upDatingFriend: {}
+    })
+
+    this.props.updateFriend(friend)
+  }
+  render() {
+    return (
+      <div className="App">
+        <Friends {...this.props} onUpdate={this.handleUpdate} />
+        <FriendForm {...this.props} />
+        {this.state.isUpdate
+          ? <FriendUpdateForm {...this.state.upDatingFriend} friendUpdate={this.updateSubmit} />
```

Comment:
`{...this.state.upDatingFriend}` is not coming in at all on `FriendUpdateForm`. Changing it to `friend={this.state.upDatingFriend}` and calling `this.props.friend...` in `FriendUpdateForm` works well.